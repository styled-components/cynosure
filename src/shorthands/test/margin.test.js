// @flow
import margin from '../margin'

describe('margin', () => {
  it('properly applies a value when passed only one', () => {
    expect(margin('12px')).toEqual({
      marginBottom: '12px',
      marginLeft: '12px',
      marginRight: '12px',
      marginTop: '12px',
    })
  })
  it('properly applies values when passed two', () => {
    expect(margin('12px', '24px')).toEqual({
      marginBottom: '12px',
      marginLeft: '24px',
      marginRight: '24px',
      marginTop: '12px',
    })
  })
  it('properly applies values when passed three', () => {
    expect(margin('12px', '24px', '36px')).toEqual({
      marginBottom: '36px',
      marginLeft: '24px',
      marginRight: '24px',
      marginTop: '12px',
    })
  })
  it('properly applies values when passed four', () => {
    expect(margin('12px', '24px', '36px', '48px')).toEqual({
      marginBottom: '36px',
      marginLeft: '48px',
      marginRight: '24px',
      marginTop: '12px',
    })
  })
  it('properly applies values when passed four', () => {
    expect(margin(12, 24, 36, 48)).toEqual({
      marginBottom: 36,
      marginLeft: 48,
      marginRight: 24,
      marginTop: 12,
    })
  })
  it('properly applies zero value', () => {
    expect(margin(0)).toEqual({
      marginBottom: 0,
      marginLeft: 0,
      marginRight: 0,
      marginTop: 0,
    })
  })
})
