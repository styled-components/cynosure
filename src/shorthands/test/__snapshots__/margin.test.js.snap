exports[`margin properly applies a value when passed only one 1`] = `
Object {
  "marginBottom": "12px",
  "marginLeft": "12px",
  "marginRight": "12px",
  "marginTop": "12px",
}
`;

exports[`margin properly applies values when passed four 1`] = `
Object {
  "marginBottom": "36px",
  "marginLeft": "48px",
  "marginRight": "24px",
  "marginTop": "12px",
}
`;

exports[`margin properly applies values when passed three 1`] = `
Object {
  "marginBottom": "36px",
  "marginLeft": "24px",
  "marginRight": "24px",
  "marginTop": "12px",
}
`;

exports[`margin properly applies values when passed two 1`] = `
Object {
  "marginBottom": "12px",
  "marginLeft": "24px",
  "marginRight": "24px",
  "marginTop": "12px",
}
`;
